File: src/main/java/com/example/aidemo/service/AIService.java
package com.example.aidemo.service;

import lombok.RequiredArgsConstructor;
import org.springframework.ai.bedrock.anthropic.AnthropicChatModel;
import org.springframework.ai.chat.ChatResponse;
import org.springframework.ai.chat.messages.UserMessage;
import org.springframework.ai.chat.prompt.Prompt;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
@RequiredArgsConstructor
public class AIService {

    private final AnthropicChatModel anthropicChatModel;

    public String generateSummary(String text) {
        String prompt = "Please summarize the following text in a concise manner: " + text;
        return generateResponse(prompt);
    }

    public String generateIdeas(String topic) {
        String prompt = "Generate 5 creative ideas related to the following topic: " + topic;
        return generateResponse(prompt);
    }

    public String improveText(String text) {
        String prompt = "Please improve the following text by making it more clear, concise, and engaging: " + text;
        return generateResponse(prompt);
    }

    public String answerQuestion(String question) {
        String prompt = "Please answer the following question in a helpful and informative way: " + question;
        return generateResponse(prompt);
    }

    private String generateResponse(String message) {
        Prompt prompt = new Prompt(List.of(new UserMessage(message)));
        ChatResponse response = anthropicChatModel.call(prompt);
        return response.getResult().getOutput().getContent();
    }
}

File: frontend/src/services/AIService.js
import axios from 'axios';

const API_BASE_URL = 'http://localhost:8080/api/ai';

class AIService {
  summarizeText(text) {
    return axios.post(`${API_BASE_URL}/summarize`, { text });
  }

  generateIdeas(topic) {
    return axios.post(`${API_BASE_URL}/ideas`, { text: topic });
  }

  improveText(text) {
    return axios.post(`${API_BASE_URL}/improve`, { text });
  }

  answerQuestion(question) {
    return axios.post(`${API_BASE_URL}/answer`, { text: question });
  }
}

export default new AIService();